src/stan/lang/grammars/semantic_actions_def.cpp

```diff
@@ -580,18 +577,51 @@ namespace stan {
         return;
       }
 
+       // check argument qualifiers
+      if (!decl.body_.is_no_op_statement()) {
+        function_signature_t decl_sig =
+          function_signatures::instance().get_definition(decl.name_, sig);
+        if (!decl_sig.first.is_ill_formed()) {
+          if (decl_sig.second.size() != arg_types.size()) {
+            error_msgs << "Declaration, definition mismatch for function "
```

Comment:
I don't like the comma here.  How about "Definition does not match declaration for function"?